third_party/terraform/utils/compute_instance_helpers.go

```diff
@@ -329,3 +329,21 @@ func flattenShieldedVmConfig(shieldedVmConfig *computeBeta.ShieldedVmConfig) []m
 		"enable_integrity_monitoring": shieldedVmConfig.EnableIntegrityMonitoring,
 	}}
 }
+
+func expandDisplayDevice(d *schema.ResourceData) *computeBeta.DisplayDevice {
+	if _, ok := d.GetOk("enable_display"); !ok {
+		return nil
+	}
+	return &computeBeta.DisplayDevice{
+		EnableDisplay:   d.Get("enable_display").(bool),
+		ForceSendFields: []string{"EnableDisplay"},
+	}
+}
+
+func extractEnableDisplay(displayDevice *computeBeta.DisplayDevice) bool {
+	if displayDevice == nil {
+		return false
```

Comment:
If the API returns an empty/nil response then you are now explicitly setting this field to `false` in the config. This will mean that if somebody doesn't set `enable_display` in their config, they will end up with `false` in their state and re-running plan will attempt to set it to `nil`, which will result in a permadiff.

You can avoid all this by either returning `nil` if it's empty or using a `Default` in the schema above. Using `Default` will explicitly send `false` to the API every time, so make sure the API is tolerant of that. I have a weak preference for just allowing `nil` as there have been some API's where there is a difference between `nil` and `false` for reasons I don't always understand.